dat/RecipesScraper/RecipesScraper/spiders/allrecipes_spider.py

```diff
@@ -0,0 +1,55 @@
+"""
+Scrape data from allrecipes.com
+"""
+import re
+import scrapy
+from RecipesScraper.items import RecipeItem
+
+
+class AllRecipesSpider(scrapy.Spider):
+    """Spider to scrape All Recipes (allrecipes.com)"""
+    name = "allrecipes"
+
+    with open("sitemap/seed_lists/allrecipes_seed_list.txt") as seedfile:
+        start_urls = ["{}?page=1".format(url.strip()) for url in seedfile.readlines()]
+
+    def parse(self, response):
+        """Parse recipe page."""
+        recipes = response.css('article.grid-col--fixed-tiles')
+        if len(recipes) > 0:
+            for recipe in recipes:
+                try:
+                    recipe_href = response.urljoin(recipe.css('a::attr(href)').extract_first())
+                    if 'video' not in recipe_href and recipe_href != response.url:
+                        yield scrapy.Request(recipe_href, callback=self.parse_recipe)
+                except AttributeError:
+                    self.log("Skipped empty article")
+
+            base_url, page_info = response.url.split("?")
+            page_number = int(page_info.split("=")[1])
+            next_page = "{}?page={}".format(base_url, page_number + 1)
+            yield scrapy.Request(next_page, callback=self.parse)
+
+    def parse_recipe(self, response):
+        """Parse the recipe to get title and ingredients."""
+        recipe_name = self.remove_non_ascii(response.css("h1.recipe-summary__h1::text").extract_first())
+        ingredients = filter(lambda ingredient: ingredient != u'Add all ingredients to list',
```

Comment:
A lot of recipe sites use either schema.org/Recipe (http://schema.org/Recipe), json+ld (http://jsonld.com/recipe/) or less frequently h recipe (http://microformats.org/wiki/h-recipe). It will be easier to have one parser that looks for those three instead of having a special parser for each site

scrapy has a library to help out with schema.org and jsonld parsing https://github.com/scrapinghub/extruct